Src/IronPython.Modules/_codecs.cs

```diff
@@ -387,24 +417,68 @@ public static PythonTuple utf_16_be_encode(CodeContext context, string input, st
 #if FEATURE_ENCODING
 
         public static PythonTuple utf_7_decode(CodeContext context, [BytesConversion]IList<byte> input, string errors = "strict", bool final = false)
-            => DoDecode(context, "utf-7", Encoding.UTF7, input, errors, final).ToPythonTuple();
+            => DoDecode(context, "utf-7", Encoding.UTF7, input, errors, NumEligibleUtf7Bytes(input, final)).ToPythonTuple();
 
         public static PythonTuple utf_7_encode(CodeContext context, string input, string errors = "strict")
             => DoEncode(context, "utf-7", Encoding.UTF7, input, errors).ToPythonTuple();
 
+        private static int NumEligibleUtf7Bytes(IList<byte> input, bool final) {
+            byte[] bytes = input as byte[] ?? ((input is Bytes bt) ? bt.GetUnsafeByteArray() : input.ToArray());
```

Comment:
Do we need to convert to an array?

Reply:
No. It was meant as performance optimization. Since the whole array has to be scanned, I thought converting to byte[] will be beneficial. But some tests I did failed to show a benefit so I will revert to enumerating IList.